app/cdap/components/WrangleHome/Components/OngoingDataExplorations/index.tsx

```diff
@@ -0,0 +1,189 @@
+    for (const connectorName of connectionsWithConnectorTypes.keys()) {
+      const values = connectionsWithConnectorTypes.get(connectorName);
+      const connections = values.map((eachValue) => ({
+        name: eachValue.name,
+        connectorType: eachValue.connectionType,
+      }));
+      connectionsWithConnectorTypeData.push(...connections);
+    }
+
+    const findConnectorType = (connection: string) => {
+      if (connection) {
+        const matchedConnection: IConnectionWithConnectorType = connectionsWithConnectorTypeData.find(
+          (eachConnection) => eachConnection.name === connection.replace('_', ' ')
+        );
+        return matchedConnection ? matchedConnection.connectorType : undefined;
+      }
+      return T.translate(
+        'features.WranglerNewUI.OnGoingDataExplorations.labels.importedDataset'
+      ).toString();
+    };
+
+    const explorationData: IExistingExplorationCard[] = [];
+
+    // Getting the workspace name, path ,workspaceId and name from MyDataPrepApi.getWorkspaceList API and
+    //  using these in params and requestBody to get Data quality from MyDataPrepApi.execute API
+
+    MyDataPrepApi.getWorkspaceList({
+      context: 'default',
+    })
+      .pipe(
+        switchMap((response: Record<string, unknown[]>) => {
+          let values: IWorkspace[] = [];
+          values = response?.values ?? [];
+
+          // sorting the workspaces based on dataset created time.
+          values.sort((a, b) => b.createdTimeMillis - a.createdTimeMillis);
+
+          const workspaces = values.map((eachValue) => {
+            const params = {
+              context: 'default',
+              workspaceId: eachValue.workspaceId,
+            };
+            const requestBody = {
+              directives: eachValue.directives,
+              limit: 1000,
+              insights: {
+                name: eachValue?.sampleSpec?.connectionName,
+                workspaceName: eachValue.workspaceName,
+                path: eachValue?.sampleSpec?.path,
+                visualization: {},
+              },
+            };
+
+            const connectorName: string | undefined = findConnectorType(
+              eachValue?.sampleSpec?.connectionName
+            );
+            if (connectorName) {
+              explorationData.push({
+                connectorType: connectorName,
+                connectionName: eachValue?.sampleSpec?.connectionName
+                  ? eachValue?.sampleSpec?.connectionName
+                  : T.translate(
+                      'features.WranglerNewUI.OnGoingDataExplorations.labels.importedDataset'
+                    ).toString(),
+                workspaceName: eachValue.workspaceName,
+                recipeSteps: eachValue.directives?.length ?? 0,
+                dataQuality: null,
+                workspaceId: eachValue.workspaceId,
+                count: 0,
+              });
+              return MyDataPrepApi.execute(params, requestBody);
```

Comment:
How does this handle 100s-1000s of workspaces in the list?

Reply:
Currently, we do not have the option to send the limit on the getWorkspaceList API,
so we are fetching all the open workspaces from the API and manipulating the response data to achieve the required output i.e. data quality.

Currently, the system will wait for all API calls to be resolved before loading the data on the page.

We can show a loader while the APIs are in progress so the user can wait till they are resolved.